storage/src/test/java/org/apache/kafka/tiered/storage/TieredStorageTestContext.java

```diff
@@ -0,0 +1,348 @@
+
+    private final Seq<KafkaBroker> brokers;
+    private final Properties producerConfig;
+    private final Properties consumerConfig;
+    private final ListenerName listenerName;
+
+    private final Serializer<String> ser;
+    private final Deserializer<String> de;
+
+    private final Map<String, TopicSpec> topicSpecs;
+    private final TieredStorageTestReport testReport;
+
+    private volatile KafkaProducer<String, String> producer;
+    private volatile KafkaConsumer<String, String> consumer;
+    private volatile Admin admin;
+    private volatile List<LocalTieredStorage> tieredStorages;
+    private volatile List<BrokerLocalStorage> localStorages;
+
+    public TieredStorageTestContext(Seq<KafkaBroker> brokers,
+                                    Properties producerConfig,
+                                    Properties consumerConfig,
+                                    ListenerName listenerName) {
+        this.brokers = brokers;
+        this.producerConfig = producerConfig;
+        this.consumerConfig = consumerConfig;
+        this.listenerName = listenerName;
+        this.ser = Serdes.String().serializer();
+        this.de = Serdes.String().deserializer();
+        this.topicSpecs = new HashMap<>();
+        this.testReport = new TieredStorageTestReport(this);
+        initContext();
+    }
+
+    private void initContext() {
+        String bootstrapServers = TestUtils.bootstrapServers(brokers, listenerName);
+        producerConfig.put(BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        // Set a producer linger of 60 seconds, in order to optimistically generate batches of
+        // records with a pre-determined size.
+        producerConfig.put(LINGER_MS_CONFIG, String.valueOf(TimeUnit.SECONDS.toMillis(60)));
+        producer = new KafkaProducer<>(producerConfig, ser, ser);
+
+        consumerConfig.put(BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        consumer = new KafkaConsumer<>(consumerConfig, de, de);
+
+        Properties adminConfig = new Properties();
+        adminConfig.put(BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        admin = Admin.create(adminConfig);
+
+        tieredStorages = TieredStorageTestHarness.getTieredStorages(brokers);
+        localStorages = TieredStorageTestHarness.getLocalStorages(brokers);
+    }
+
+    public void createTopic(TopicSpec spec) throws ExecutionException, InterruptedException {
+        NewTopic newTopic;
+        if (spec.getAssignment() == null || spec.getAssignment().isEmpty()) {
+            newTopic = new NewTopic(spec.getTopicName(), spec.getPartitionCount(), (short) spec.getReplicationFactor());
+        } else {
+            Map<Integer, List<Integer>> replicasAssignments = spec.getAssignment();
+            newTopic = new NewTopic(spec.getTopicName(), replicasAssignments);
+        }
+        newTopic.configs(spec.getProperties());
+        admin.createTopics(Collections.singletonList(newTopic)).all().get();
```

Comment:
same as above -- creates the Admin instance on each invocation. 

To write complex tests, we need the same instance for multiple actions. Creating it for each action might be expensive and difficult to read through the logs.

Reply:
Used `TestUtils.waitForAllPartitionsMetadata` instead of `TestUtils.createTopicWithAdmin` as the latter requires converting the replicaAssignment Map from java to scala. Will do it later once the PR is stable.